process_scheduling_algorithms/preemptive_priority_scheduling.c

```diff
@@ -0,0 +1,158 @@
+/**
+ * @file
+ * @brief
+ * [Preemptive Priority
+ * Scheduling]
+ *  Preemptive Priority Scheduling is a CPU scheduling algorithm
+    that assigns priorities to processes and selects the process
+    with the highest priority to execute. If a higher-priority
+    process becomes available, it preempts the currently executing
+    process. This algorithm ensures that higher-priority processes
+    are given preference, but it may lead to starvation of lower-priority
+    processes if not handled properly.*
+ *
+ * @details
+ Algorithm Overview:
+    1. Initialize a set of processes with attributes such as ID, arrival time,
+       burst time, priority, executed time, and completion time.
+    2. Start with an empty queue of ready processes and initialize variables
+       to keep track of the current time, executed processes, and the currently
+       executing process.
+    3. Repeat the following steps until all processes are executed:
+       a. Find the process with the highest priority that has arrived and
+          not completed yet.
+       b. Execute the selected process for one time unit, updating its
+          executed time and the current time.
+       c. If the selected process completes, calculate its completion time,
+          print a completion message, and update turnaround and waiting times.
+       d. If no suitable process is found to execute, increment the current time.
+    4. Calculate and print the average turnaround time and average waiting time.
+ * @author [Archana Saroj](https://github.com/SarojArchana)
+ */
+#include <stdio.h>    /// for IO operations (`printf`)
+/**
+ * @brief Structure to represent a process
+ */
+struct Process {
+    int id;             /**< Process ID */
+    int arrival_time;   /**< Arrival time of the process */
+    int burst_time;     /**< Burst time of the process */
+    int priority;       /**< Priority of the process */
+    int executed_time;  /**< Time the process has been executed */
+    int completion_time;/**< Completion time of the process */
```

Comment:
```suggestion
    int id;             ///< Process ID
    int arrival_time;   ///< Arrival time of the process 
    int burst_time;     ///< Burst time of the process
    int priority;       ///< Priority of the process
    int executed_time;  ///< Time the process has been executed
    int completion_time;///< Completion time of the process 
```
Since these are single line no need for them to be multiline